File: Website/AdminSite/service/authenticate/logoutService.js
export function logoutService() {
  fetch("http://localhost:8083/cinema/logout", {
    method: "POST",
    headers: {
      "Content-Type": "application/json",
    },
    body: JSON.stringify({
      token: sessionStorage.getItem("token"),
    }),
  })
    .then((response) => {
      if (!response.ok) {
        throw new Error("Network response was not ok");
      }
      return response.json();
    })
    .then((data) => {
      sessionStorage.clear();
    })
    .catch((error) => {
      console.error("Error LOGOUT:", error);
    });
}

File: App/movie_booking_app/lib/services/Users/logout/logout_service.dart
import 'dart:convert';
import 'dart:io';
import 'package:flutter/material.dart';
import 'package:flutter_dotenv/flutter_dotenv.dart';
import 'package:http/http.dart' as http;
import 'package:movie_booking_app/modules/valid/show_message.dart';
import 'package:movie_booking_app/provider/shared-preferences/prefs.dart';

class LogOutServices {
  static Future<dynamic> logout(context) async {
    Preferences pref = Preferences();
    String? token = await pref.getTokenUsers();
    try {
      final getURL = dotenv.env['LOG_OUT']!;
      final url = getURL;
      final logoutBody = json.encode({'token': token});

      final response = await http.post(
        Uri.parse(url),
        headers: {
          'Content-Type': 'application/json',
        },
        body: logoutBody,
      );

      if (response.statusCode != 200) {
        debugPrint('Log out Service Code: ${response.statusCode}');
      }
      final responseData = json.decode(response.body);

      if (responseData['code'] != 1000) {
        await pref.clear();
        return;
      }
      return responseData['message'];
    } on SocketException {
      ShowMessage.noNetworkConnection(context);
    } catch (err) {
      ShowMessage.unExpectedError(context);
    }
  }
}
